monai/networks/layers/convutils.py

```diff
@@ -93,11 +97,74 @@ def gaussian_1d(sigma: Union[float, torch.Tensor], truncated: float = 4.0) -> np
         1D torch tensor
 
     """
-    sigma = torch.as_tensor(sigma).float()
-    if sigma <= 0 or truncated <= 0:
+    sigma = torch.as_tensor(sigma, dtype=torch.float)
+    if sigma <= 0.0 or truncated <= 0.0:
         raise ValueError(f"sigma and truncated must be positive, got {sigma} and {truncated}.")
     tail = int(sigma * truncated + 0.5)
-    x = torch.arange(-tail, tail + 1).float()
-    t = 1 / (torch.tensor(2.0).sqrt() * sigma)
-    out = 0.5 * ((t * (x + 0.5)).erf() - (t * (x - 0.5)).erf())
-    return out.clamp(min=0)
+    if approx.lower() == "simple":
+        x = torch.arange(-tail, tail + 1, dtype=torch.float)
+        t = 1.0 / (torch.tensor(2.0).sqrt() * sigma)
+        out = 0.5 * ((t * (x + 0.5)).erf() - (t * (x - 0.5)).erf())
+        return out.clamp(min=0)
+    if approx.lower() == "refined":
+        out = [_modified_bessel_0(sigma), _modified_bessel_1(sigma)]
+        while len(out) <= tail:
+            out.append(_modified_bessel_i(len(out), sigma))
+        ans = out[:0:-1]
+        ans.extend(out)
+        ans = torch.stack(ans) * torch.exp(-sigma)
+        ans /= ans.sum()
```

Comment:
Also, I don't think we should normalise to sum up to one. If the truncation is large, it should indeed sum up to less than one. At least, maybe the normalisation to one should be an option available for all kernel implementations?